### PR TITLE
Support importing native modules in native packages.

### PR DESCRIPTION
**Pull request by tannewt**
This only fixes the `import` portion. It doesn't actually change
reference behavior because modules within a package could already
be referenced through the parent package even though an error should
have been thrown.

### CODE REVIEW

**Comment by tannewt** [comment]
@jepler @v923z This should allow for importing submodules of `ulab`.

I've linked @jimmo to it and I bet @dpgeorge would be interested too.

**Comment by v923z** [comment]
So the claim is that 
```
from ulab import fft

fft.fft(...)
```

would work with this fix, right? If so, it is really great, thanks for taking care of it! 

@dpgeorge Damien, do you think this could be merged into `micropython` proper on a reasonable time scale? With that we could maintain a single version of `ulab`, ant that would save some resources on our end.

**Comment by dpgeorge** [comment]
@tannewt thanks for the note, I'll look into incorporating this or similar.  I didn't yet look in detail but this might be related: https://github.com/micropython/micropython/pull/4731

**Comment by jimmo** [comment]
See https://github.com/micropython/micropython/pull/5701 for the equivalent change upstream. Thanks for suggesting this, I think it's a much better approach than micropython#4731

Some minor notes on this PR:
 - Might need to check when looking at the globals dict that the result is actually a module (i.e. type == mp_module_type).
 - You can avoid the change in objmodule.c by just setting a flag in builtinimport, then avoid calling mp_store_attr at all.

**Comment by tannewt** [comment]
Thanks @jimmo and @dpgeorge. Will keep an eye on upstream and plan on  replacing this PR next time we merge from upstream.

@jimmo Good idea on the type check. I've added it and confirmed we can't `import microcontroller.cpu` which is not a module. My preference is to keep the set_attr because I'm assuming a module will track whether it's members have been imported in the future rather than having import manage it.

**Comment by tannewt** [comment]
@jepler I think this is ready to merge.

File: py/gc_long_lived.c [hunk 1]
```diff
@@ -85,11 +85,13 @@ mp_obj_property_t *make_property_long_lived(mp_obj_property_t *prop, uint8_t max
     return gc_make_long_lived(prop);
 }
 
+extern const mp_obj_dict_t mcu_module_globals;
+
```
Comment (jepler) [inline py/gc_long_lived.c]: this added extern declaration may not be needed, I don't see where it is used.

Comment (tannewt) [reply py/gc_long_lived.c]: Yup, I added it for debugging iirc. Thanks!

**Comment by tannewt** [comment]
@jepler Removed! Please take another look!

**Pull request merged**